IQSS/dataverse-frontend
# 347 Add eslint plugin react hooks

## What this PR does / why we need it:
Adds Eslint plugin for React Hooks.
We need it to enforce [Rules of Hooks](https://react.dev/reference/rules/rules-of-hooks).

## Which issue(s) this PR closes:

- Closes https://github.com/IQSS/dataverse-frontend/issues/347

## Special notes for your reviewer:
The plugin raised only a few **_errors_** when initially configured, only related to calling hooks conditionally, all were resolved.
There are several minor _**warnings**_ that we could fix as we come across them.
I only hide via a comment of eslint two errors regarding the use of a hook in the Story of the DatasetAlert as I think it might take more time than expected and it is just a story, however we could see how to configure the alerts story to avoid the eslint error in the future.
## Suggestions on how to test this:
- Run tests to check everything works as usual.
- Locate a useEffect call after a conditional statement in a react component, it should show an eslint error.
- Use a local state variable or a prop inside a useEffect and don't add it to the dependency array of it, it should show an eslint warning.
## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
No
## Is there a release notes update needed for this change?:
No
## Additional documentation:
